src/ejabberd_s2s_out.erl

```diff
@@ -130,6 +130,22 @@
 
 -define(SOCKET_DEFAULT_RESULT, {error, badarg}).
 
+
+-define(CLOSE_GENERIC(StateName, Reason, StateData),
+    ?LOG_INFO(#{what => s2s_out_closing,
+                text => <<"Closing s2s connection">>,
+                state_name => StateName, reason => Reason,
+                myname => StateData#state.myname, server => StateData#state.server}),
+    {stop, normal, StateData}).
+
+-define(CLOSE_GENERIC(StateName, Reason, El, StateData),
+    ?LOG_INFO(#{what => s2s_out_closing,
+                text => <<"Closing s2s connection">>,
+                exml_packet => El,
+                state_name => StateName, reason => Reason,
+                myname => StateData#state.myname, server => StateData#state.server}),
+    {stop, normal, StateData}).
+
```

Comment:
If the only difference is containing the element, perhaps this could be done with just one macro that takes a map and merges it with the default one? Whichever is easier and more readable.

Reply:
Keep it simple.